codec/decode_fields.go

```diff
@@ -193,7 +194,18 @@ func DecodeLengthDelimitedField(fd *desc.FieldDescriptor, bytes []byte, mf Messa
 	case fd.GetType() == descriptor.FieldDescriptorProto_TYPE_MESSAGE ||
 		fd.GetType() == descriptor.FieldDescriptorProto_TYPE_GROUP:
 		msg := mf.NewMessage(fd.GetMessageType())
-		err := proto.Unmarshal(bytes, msg)
+
+		opts := protoV2.UnmarshalOptions{
```

Comment:
Why is this package changed? This is a serious smell regarding the scope of this PR.